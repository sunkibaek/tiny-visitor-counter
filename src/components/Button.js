import { Pressable, StyleSheet, Text } from "react-native";

const styles = StyleSheet.create({
  baseContainer: {
    height: 60,
    alignItems: "center",
    justifyContent: "center",
    paddingHorizontal: 16,
    paddingVertical: 8,
  },
  baseText: {
    fontSize: 16,
    fontWeight: "600",
  },
  defaultContainer: {
    backgroundColor: "#141414",
  },
  defaultText: {
    color: "#ffffff",
  },
  linkContainer: {
    backgroundColor: "transparent",
  },
  linkText: {
    color: "#141414",
  },
  fullWidthContainer: {
    flex: 1,
  },
});

const BaseButton = ({
  title,
  onPress,
  containerStyle,
  textStyle,
  fullWidth,
}) => (
  <Pressable
    onPress={onPress}
    style={[
      styles.baseContainer,
      fullWidth && styles.fullWidthContainer,
      containerStyle,
    ]}
  >
    <Text style={[styles.baseText, textStyle]}>{title}</Text>
  </Pressable>
);

const Button = ({ title, onPress, fullWidth }) => (
  <BaseButton
    onPress={onPress}
    containerStyle={styles.defaultContainer}
    textStyle={styles.defaultText}
    title={title}
    fullWidth={fullWidth}
  />
);

Button.Link = ({ title, onPress, fullWidth }) => (
  <BaseButton
    title={title}
    onPress={onPress}
    containerStyle={styles.linkContainer}
    textStyle={styles.linkText}
    fullWidth={fullWidth}
  />
);

export default Button;
