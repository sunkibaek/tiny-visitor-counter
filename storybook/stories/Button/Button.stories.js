import { action } from "@storybook/addon-actions";
import { boolean, text } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react-native";
import React from "react";
import { View } from "react-native";

import Button from "../../../src/components/Button";
import CenterView from "../CenterView";

storiesOf("Button", module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add("default", () => (
    <Button
      onPress={action("clicked-text")}
      title={text("Button text", "Default Button")}
    />
  ))
  .add("with fullWidth (in row)", () => (
    <View style={{ flexDirection: "row" }}>
      <Button
        onPress={action("clicked-text")}
        title={text("title", "Button 1")}
        fullWidth={boolean("fullWidth", true)}
      />

      <Button
        onPress={action("clicked-text")}
        title={text("title", "Button 2")}
        fullWidth={boolean("fullWidth", true)}
      />
    </View>
  ));

storiesOf("Button.Link", module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add("default", () => (
    <Button.Link
      onPress={action("clicked-text")}
      title={text("title", "Link Button")}
    />
  ));
