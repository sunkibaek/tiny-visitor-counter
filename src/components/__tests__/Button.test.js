import { render, screen, fireEvent } from "@testing-library/react-native";

import Button from "../Button";

describe("Button", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe("with title", () => {
    it("renders correctly", () => {
      render(<Button title="Test Button" />);

      expect(screen.toJSON()).toMatchSnapshot();
    });

    it("renders title", () => {
      const titleText = "Test Button";

      render(<Button title={titleText} />);

      const title = screen.getByText(titleText);
      expect(title.props.children).toBe(titleText);
    });
  });

  describe("with onPress", () => {
    it("calls onPress when pressed", () => {
      const onPress = jest.fn();
      const titleText = "Test Button";
      render(<Button onPress={onPress} title={titleText} />);

      const title = screen.getByText(titleText);
      fireEvent.press(title);

      expect(onPress).toBeCalledTimes(1);
    });
  });
});
