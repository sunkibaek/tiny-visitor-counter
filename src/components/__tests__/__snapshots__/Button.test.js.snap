// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button with title renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 60,
        "justifyContent": "center",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      },
      undefined,
      Object {
        "backgroundColor": "#141414",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 16,
          "fontWeight": "600",
        },
        Object {
          "color": "#ffffff",
        },
      ]
    }
  >
    Test Button
  </Text>
</View>
`;
